iceoryx_utils/test/moduletests/test_unit_duration.cpp

```diff
@@ -18,218 +18,686 @@
+    auto time1 = 1_s + 300_us;
+    auto time2 = 1_s + 200_us;
+    auto time3 = 400_us;
+    auto time4 = 100_us;
+    EXPECT_FALSE(time2 >= time1);
+    EXPECT_FALSE(time3 >= time1);
+    EXPECT_FALSE(time3 >= time2);
+    EXPECT_FALSE(time4 >= time3);
 }
 
-TEST(Duration_test, multiplyDuration)
+// END COMPARISON TESTS
+
+// BEGIN ARITHMETIC TESTS
+
+TEST(Duration_test, AddDuration)
 {
-    auto time1 = 5.2_s;
+    auto time1 = 5_s + 200_ms;
     auto time2 = 200_ms;
-    auto time3 = 300000_us;
+    auto time3 = 700000_us;
 
-    auto result = time1 * time2;
-    EXPECT_NEAR(result.seconds<double>(), 1.04, tenTimesEpsilon);
-    result = result * time3;
-    EXPECT_NEAR(result.seconds<double>(), 0.312, tenTimesEpsilon);
+    EXPECT_THAT(time1.milliSeconds<uint64_t>(), Eq(5200U));
 
-    auto time4 = 10_s;
+    auto result = time1 + time2;
+    EXPECT_EQ(result.milliSeconds<uint64_t>(), 5400U);
 
-    result = result * time4;
-    EXPECT_NEAR(result.seconds<double>(), 3.12, tenTimesEpsilon);
+    result = result + time3;
+    EXPECT_EQ(result.microSeconds<uint64_t>(), 6100000U);
 }
 
-TEST(Duration_test, divideDuration)
+TEST(Duration_test, SubtractDuration)
 {
-    auto time1 = 5.2_s;
+    auto time1 = 6_s - 800_ms;
     auto time2 = 200_ms;
     auto time3 = 300000_us;
 
-    auto result = time1 / time2;
-    EXPECT_NEAR(result.seconds<double>(), 26, tenTimesEpsilon);
-    result = result / time3;
-    EXPECT_NEAR(result.seconds<double>(), 86.666666666666666, tenTimesEpsilon);
+    EXPECT_THAT(time1.milliSeconds<uint64_t>(), Eq(5200U));
+
+    auto result = time1 - time2;
+    EXPECT_EQ(result.milliSeconds<uint64_t>(), 5000U);
+
+    result = result - time3;
+    EXPECT_EQ(result.microSeconds<uint64_t>(), 4700000U);
 
     auto time4 = 10_s;
 
-    result = result / time4;
-    EXPECT_NEAR(result.seconds<double>(), 8.666666666666666, tenTimesEpsilon);
+    result = result - time4;
+    EXPECT_EQ(result.milliSeconds<uint64_t>(), 0U);
 }
 
-TEST(Duration_test, constructFromTimespec)
+TEST(Duration_test, MultiplyDuration)
 {
-    struct timespec value;
-    value.tv_sec = 123;
-    value.tv_nsec = 456;
+    auto time = 5_s + 800_ms;
 
-    Duration result(value);
-    EXPECT_EQ(result.nanoSeconds<uint64_t>(), 456ull + 1000000000ull * 123ull);
-}
+    EXPECT_THAT(time * 2, Eq(11_s + 600_ms));
+    EXPECT_THAT(time * 2U, Eq(11_s + 600_ms));
+    EXPECT_THAT(time * 2.95, Eq(17_s + 110_ms));
 
-TEST(Duration_test, constructFromTimeval)
-{
-    struct timeval value;
-    value.tv_sec = 1337;
-    value.tv_usec = 42;
+    EXPECT_THAT(2 * time, Eq(11_s + 600_ms));
+    EXPECT_THAT(2U * time, Eq(11_s + 600_ms));
+    EXPECT_THAT(2.95 * time, Eq(17_s + 110_ms));
 
-    Duration result(value);
-    EXPECT_EQ(result.microSeconds<uint64_t>(), 42 + 1000000 * 1337);
+    EXPECT_THAT(time * -1, Eq(0_s));
+    EXPECT_THAT(time * -1.0, Eq(0_s));
 }
 
-TEST(Duration_test, isZeroWhenConstructedFromNegativeChronoMilliSeconds)
+TEST(Duration_test, DivideDuration)
```

Comment:
We should test:
- division by zero!
- division with negative number
- division with 0.0000000001f and a large duration type so that you run into an overflow

Reply:
can't do, there is no `operator/` for `Duration` 😁